This New branch is building upon the sqlite_test PR as it passed the testsuite

Farreeda opened the pull request.

TheCedarPrince committed:
Attempt to fix JDBC dispatch for DBInterface

Farreeda committed:
Creating test for _dbconnect function of SQLite

fixing the file path issue by identifying its type to String

- testing it(Passed)

Commit:
fixing the file_path issue

Commit:
.

Commit:
updating PR

removed .vscode folder
added

Commit:
with github secrets

Commit:
back to original

Commit:
.

Commit:
updating the path

Commit:
Added quering

Commit:
removing database in the test

Commit:
updates

Commit:
Adding Dataframes and examples

Commit:
.

Commit:
.

Commit:
allowing the test to access the function for the workflow

Commit:
add / for absolute path :)

Commit:
.

Commit:
.

Commit:
.

Commit:
path problems

Commit:
.

Commit:
.

Commit:
testing mysql and postgresql #1

testing the github secrets

Commit:
initial documentation

TheCedarPrince committed:
Cleaned workflow syntax

TheCedarPrince committed:
Removed test dependencies and put them into test dependencies

TheCedarPrince committed:
Added author names and comapt and removed test dependencies

TheCedarPrince committed:
Deprecated ODBC JDBC and their dispatch

TheCedarPrince committed:
Removed ODBC JDBC functions

TheCedarPrince committed:
Updated test suite to work properly

TheCedarPrince committed:
Added test dependenices

TheCedarPrince committed:
Removed invalidations workflow temporarily

TheCedarPrince committed:
Exploring why pull request is not triggering actions

TheCedarPrince committed:
Added proper variable naming

TheCedarPrince committed:
Removed API reference

TheCedarPrince committed:
Testing for API error

TheCedarPrince committed:
Trying to fix up API error

TheCedarPrince committed:
Added necessary import for the API

TheCedarPrince committed:
Removed import as not needed

TheCedarPrince committed:
Included LibPQ fork until PR is added from LibPQ

TheCedarPrince committed:
Added authors

TheCedarPrince committed:
Redid postgresql interface

TheCedarPrince committed:
Finalized tests at the moment

TheCedarPrince committed:
Add manifest for test environment

TheCedarPrince committed:
Just testing without MySQL server support

TheCedarPrince committed:
Added back in MySQL tests

TheCedarPrince commented on .github/workflows/ci.yml:
Had to change the syntax to get this to re-trigger but it is now working.

TheCedarPrince commented on Project.toml:
Updated authors and compat entries.

TheCedarPrince commented on src/DBConnector.jl:
I overhauled this a little to deprecate some support for ODBC and JDBC connections as well as add a proper import.

TheCedarPrince commented on src/mysql.jl:
I wasn't able to get tests for thsi working so I am not sure how well this works.

TheCedarPrince commented on src/postgresql.jl:
I know we were thinking that your function `_dbconnect` could have kwargs, but I think I'd rather have the user have to define their own `kws` to handle the specifics of the connection string like in the original _dbconnect that I brought back. What do you think @Farreeda ?

TheCedarPrince replied:
Also, I was able to test the old version and that does work on my computer to run tests but I couldn't get the new dispatch working that you had created.

Farreeda replied:
If that means adding kwargs in addition to arguments, this is a good idea. No connection can happen without the identified arguments as user, password, host, do you agree @TheCedarPrince ?
which dispatch?

TheCedarPrince commented on src/sqlite.jl:
This works perfectly!

TheCedarPrince commented on test/Project.toml:
Added required test dependencies.

TheCedarPrince commented on test/runtests.jl:
I could not get MySQL tests working on the GitHub action but I know it is due to the connection string... Could you attempt debugging this?

TheCedarPrince commented on test/runtests.jl:
```diff
@@ -1,4 +1,36 @@
+
+end
+
+@testset "_dbconnect function for LibPQ" begin
+
+    conn= DBConnector._dbconnect(LibPQ.Connection,  host = ENV["POSTGRES_HOST"], user = ENV["POSTGRES_USER"], dbname = "mimic", password = ENV["POSTGRES_PASSWORD"])
+    @test @isdefined conn
+
+end
```
Could you make sure the tests work? I think the host string needs to be formatted again or something... Not entirely sure. What you can do for testing is try connecting to the database on the server from your local machine and establishing how the connection string should work.

TheCedarPrince reviewed (changes requested):
I think this is close with GitHub Actions being nearly done. That said, the tutorials look great and the tests look solid. I think the biggest issue is just connecting to the server. Otherwise, great work. Let me know if you have any questions or issues.

Commit:
fixing tests

Commit:
testing the LibPQ test

Commit:
.

Commit:
.

Commit:
.

Commit:
.

Commit:
fixing postgres function

Commit:
checking it is not env error

Commit:
again

TheCedarPrince commented on test/runtests.jl:
```diff
@@ -1,4 +1,38 @@
+
+@testset "_dbconnect function for LibPQ" begin
+
+    #conn= DBConnector._dbconnect(LibPQ.Connection,  ENV["POSTGRES_HOST"],ENV["POSTGRES_USER"], ENV["POSTGRES_PASSWORD"], "omop")
+    conn = DBConnector._dbconnect(LibPQ.Connection,  "199.180.155.65", "postgres", "postgres3", "omop")
+    @test @isdefined conn
```
Hey Fareeda, could you change these strings to the appropriate GitHub Environment variables?

TheCedarPrince commented:
Hey @Farreeda, once you are done with the last comment I made to make sure the LibPQ tests still work even with GitHub environment variables, I think we can go ahead and merge this even though we don't have the test suite set-up for the MySQL server. Thanks!

Commit:
sql same

so the problem is with github secrets variables, let's test on mySQL

Commit:
kwargs for mysql and postgresql

Almost the secrets issue is because the PR has no access to the environment secrets of the base repo
